bellman/src/domain/mod.rs

```diff
@@ -0,0 +1,342 @@
+pub mod arith;
+pub mod linear;
+pub mod fft;
+pub mod multiexp;
+
+pub use arith::*;
+pub use linear::*;
+pub use fft::*;
+pub use multiexp::*;
+
+use ff::{Field, PrimeField, ScalarEngine};
+
+use crate::{error, multicore};
+use error::{SynthesisError, Result};
+use multicore::Worker;
+use arith::{Scalar, Group};
+
+use std::ops;
+
+/// A `Domain` abstraction for
+/// performing various kinds of polynomial arithmetic on top of
+/// the scalar field.
+///
+/// In pairing-based SNARKs like Groth16, we need to calculate
+/// a quotient polynomial over a target polynomial with roots
+/// at distinct points associated with each constraint of the
+/// constraint system. In order to be efficient, we choose these
+/// roots to be the powers of a 2^n root of unity in the field.
+/// This allows us to perform polynomial operations in O(n)
+/// by performing an O(n log n) FFT over such a domain.
```

Comment:
This is very good (excellent) documentation. This should be the standard for the rest of the repo.

Reply:
Agreed, this is their documentation. 

I would like to avoid docs for our work until we push out a stable release per clean code. There’s already a lot of dead comments and I don’t want to add more. 

What’re your thoughts here?

Reply:
Agreed, as long as we're conscious of the necessity to do so